README.md

```diff
@@ -76,13 +76,16 @@ change based on the current project/directory, as well as with different argumen
 
 Profiles are defined with the following fields:
 
-* `arch` The architecture (only amd64 or arm64 supported currently) to run the image as.
+* `arch` The architecture (`amd64`, `arm64`, `386`, `arm`, or `arm/v6`) to run the image as.
 	- Note the architecture does NOT have to match the host in most cases where emulation is set up. See [Emulation](#emulation) below
 	- Defaults to the detected current architecture.
 * `image` The docker image used by this profile. Can be overriden by `-image`
```

Comment:
like the main debian:bookworm will autodetect architectures from this list

![image](https://github.com/viamrobotics/canon/assets/7256523/8e4dda8b-e91f-4e79-97f1-8f975eeb7772)

Reply:
Yeah, they had that before I ever wrote canon. But it works... poorly. Manifests only exist and are only understood at download time. When requesting an image that is actually a "manifest" it basically just has an extra step to redirect to the "real" image based on the platform. The problem is that once that's downloaded, that _specific_ image (e.g. just that one architecture) is tagged with the manifest tag. E.g. "debian:bookworm" So if you then go to start "debian:bookworm" with a different platform after that, it warns that it can't find the image locally, and has to download a new one. Then if you switch platforms back, it happens again. Thankfully the download process itself recognizes the layers in the image as already being cached locally, so doesn't really download much, but it still has to check them each time and attempt it. And if you're offline when switching platforms... you can't. Unless you happen to remember the sha of the actual image you want and use that instead of the tag.

So allowing images to be specified directly per architecture keeps switching much faster, and able to work offline too. If you want to use the manifest feature though, you can always just specify the manifest image as "image" (without any specific architecture) and it'll get used for all platforms. I added code to make canon detect the platform mismatch and re-download in those cases, just like the docker cli tool does (as the support isn't in the library itself.)

```
otterverse@arclight ~ $ docker run --rm -it --platform linux/arm64 debian:bookworm
root@f3dfc28ee245:/# uname -a
Linux f3dfc28ee245 6.4.7-arch1-1 #1 SMP PREEMPT_DYNAMIC Thu, 27 Jul 2023 22:02:18 +0000 armv7l GNU/Linux
root@f3dfc28ee245:/# 
exit
otterverse@arclight ~ $ docker run --rm -it --platform linux/amd64 debian:bookworm
Unable to find image 'debian:bookworm' locally
bookworm: Pulling from library/debian
Digest: sha256:eaace54a93d7b69c7c52bb8ddf9b3fcba0c106a497bc1fdbb89a6299cf945c63
Status: Downloaded newer image for debian:bookworm
root@6acaa491f1b8:/# uname -a
Linux 6acaa491f1b8 6.4.7-arch1-1 #1 SMP PREEMPT_DYNAMIC Thu, 27 Jul 2023 22:02:18 +0000 x86_64 GNU/Linux
root@6acaa491f1b8:/# 
exit
otterverse@arclight ~ $ docker run --rm -it --platform linux/arm64 debian:bookworm
Unable to find image 'debian:bookworm' locally
bookworm: Pulling from library/debian
Digest: sha256:eaace54a93d7b69c7c52bb8ddf9b3fcba0c106a497bc1fdbb89a6299cf945c63
Status: Downloaded newer image for debian:bookworm
root@2f99667e0ef7:/# uname -a
Linux 2f99667e0ef7 6.4.7-arch1-1 #1 SMP PREEMPT_DYNAMIC Thu, 27 Jul 2023 22:02:18 +0000 aarch64 GNU/Linux
root@2f99667e0ef7:/# 
exit
```